pkg/util/vdbutil/parser.go

```diff
@@ -61,23 +61,54 @@ func ParseDataSourcesInfoFromDdl(ddl string) []DatasourceInfo {
 	return sources
 }
 
-// HasMaterializationTags --
-func HasMaterializationTags(ddl string) bool {
-	materialization := "MATERIALIZED\\s+'TRUE'"
+// MaterializiedViewsInDdl --
```

Comment:
I thought we were going to opt for less, not more parsing?  Just a flag to indicate datagrid usage.  We should do this sparingly as comments and in this case new lines will disrupt this parsing - it's also not fully contextually (options clause) aware.

Reply:
Can you give an inline comment example? I need the FQN of the view names to manage the cache when the Infinispan moves to support Cache as CR, that is is the reason for additional parsing. Otherwise, there are no other good options.

We could start writing yacc based parser in Go as defined here https://github.com/sougou/parser_tutorial or handwritten ones like in here https://blog.gopheracademy.com/advent-2014/parsers-lexers/

Reply:
> Can you give an inline comment example?

It's not really about a single comment, it's the complexity that it introduces to these regex expressions - anywhere you can have whitespace you can have a comment - and multi-line comments I believe would be pretty messy to capture.

>  need the FQN of the view names to manage the cache when the Infinispan moves to support Cache as CR, that is is the reason for additional parsing. Otherwise, there are no other good options.

Options:

- introduce a fuller parser to the operator.  Problems still exist as you are doing a lot of contextual parsing here - assumptions about what schema we're creating things under for example.  If we add support for create view schema.name - then that would break this.

- Move the crd creation to Teiid Spring Boot where we are currently implicitly creating the caches.  If we don't do this, we'll need to set the translator to not create if a cache does not exist.

- Use an explicit cr construct:
```
  datagrid:
     - viewname1: basecachename
     - viewname2 ...
```
where the view name would need to be fully qualified and base cache name would be manipulated to account for multiple deployments.

- Or require something similar in DDL - by reusing the MATERIALIZATION_TARGET property (as an unqualfied name) or a new property that indicates the cache name to use.  That simplifies the parsing to just that property key/value.  This would require some additional logic in the codegen plugin more than likely.

- Have the operator run a Java "vdb service" pod that provides a rest interface for getting vdb information from submitted ddl.

A general thought is that we'll need to default to transactional caches.

I'd also like to have it spelled out the various creation / deletion scenarios we are proposing to support.  It looks like so far we have:

- teiid-cache-store indicates a cache store to be used by all dv in the namespace
- vdb-cache-store indicates a cache store to be used by all deployments of the given vdb
???

Maybe a table of who creates/destroys the secret, the cluster, and the caches.